### PR TITLE
 [feat][doc] show how to use Pulsar BOM with Java client

### PR DESCRIPTION
**Pull request by onobc**
This code proposal adds a section to the Pulsar Java Client reference docs which shows how to use the [Pulsar BOM](https://github.com/apache/pulsar/blob/master/pip/pip-326.md) when setting up the Java client. The BOM ensures that all Pulsar dependencies (both direct and transitive) are at the same expected version.

Here is a motivating discussion which is the catalyst for this improvement: https://github.com/apache/pulsar/discussions/22328

### ✅ Contribution Checklist

- [x] I read the [contribution guide](https://pulsar.apache.org/contribute/document-contribution/)
- [x] I updated the [versioned docs](https://pulsar.apache.org/contribute/document-contribution/#update-versioned-docs)

### CODE REVIEW

File: versioned_docs/version-3.2.x/client-libraries-java-setup.md [hunk 1]
```diff
@@ -42,10 +42,53 @@ If you use Gradle, add the following information to the `build.gradle` file.
 def pulsarVersion = '@pulsar:version@'
 
 dependencies {
-    compile group: 'org.apache.pulsar', name: 'pulsar-client', version: pulsarVersion
+    implementation "org.apache.pulsar:pulsar-client:${pulsarVersion}"
```
Comment (onobc) [inline versioned_docs/version-3.2.x/client-libraries-java-setup.md:45]: **[UNRELATED]** updated the configuration to `implementation` and uses the short form of the dependency.

File: docs/client-libraries-java-setup.md [hunk 2]
```diff
@@ -42,10 +42,53 @@ If you use Gradle, add the following information to the `build.gradle` file.
 def pulsarVersion = '@pulsar:version@'
 
 dependencies {
-    compile group: 'org.apache.pulsar', name: 'pulsar-client', version: pulsarVersion
+	implementation "org.apache.pulsar:pulsar-client:${pulsarVersion}"
```
Comment (onobc) [inline docs/client-libraries-java-setup.md:45]: **[UNRELATED]** updated the configuration to `implementation` and uses the short form of the dependency.

File: versioned_docs/version-3.2.x/client-libraries-java-setup.md [hunk 3]
```diff
@@ -42,10 +42,53 @@ If you use Gradle, add the following information to the `build.gradle` file.
...
 }
 ```
 
+### Pulsar BOM
```
Comment (onobc) [inline versioned_docs/version-3.2.x/client-libraries-java-setup.md:49]: I did not see an easy way to not violate DRY principle here and factor this section out and use it across main and 3.2x versioned docs. Even if we Markdown supported including other Markdown files (which I dont think it does) we would still have the cross-version to think about. For now it is duplicated in 2 places.

Comment (lhotari) [reply versioned_docs/version-3.2.x/client-libraries-java-setup.md:49]: Yes, the solution isn't "clean" in that sense.

**Review by lhotari — APPROVED** [review]
LGTM. Thanks for the contribution @onobc !

**Pull request merged**